pkg/operation/botanist/controlplane/kubeapiserver/deployment/kube_apiserver.go

```diff
@@ -0,0 +1,761 @@
+	// defaultAPIServerReplicas is the default replicas for the kube-apiserver
+	defaultAPIServerReplicas = 1
+	// konnectivityUDSName is the Unix Domain Socket name used for communication of the Shoot API server to
+	// konnectivity server sidecar for traffic that meant for components in the Shoot VPC.
+	konnectivityUDSName = "konnectivity-server.socket"
+	// portNameHTTPS is a constant for the name of the HTTPS port of the kube-apiserver.
+	portNameHTTPS = "https"
+	// vpnClientPort is a constant for the TCP tunnel port of the vpn client.
+	vpnClientPort = int32(1194)
+	// vpnPort is a constant for the OpenVPN port of the vpn server in the Shoot.
+	vpnPort = "4314"
+
+	// constants for CA bundle mounts
+	volumeMountNameCABundleFedoraRHEL6Openelec = "fedora-rhel6-openelec-cabundle"
+	volumeMountPathCABundleFedoraRHEL6Openelec = "/etc/pki/tls"
+	volumeMountNameCABundleCentOSRHEl7         = "centos-rhel7-cabundle"
+	volumeMountPathCABundleCentOSRHEl7Dir      = "/etc/pki/ca-trust/extracted/pem"
+	volumeMountNameCABundleEtcSSL              = "etc-ssl"
+	volumeMountPathCABundleEtcSSL              = "/etc/ssl"
+	volumeMountNameCABundleUsrShareCacerts     = "usr-share-cacerts"
+	volumeMountPathCABundleUsrShareCacerts     = "/usr/share/ca-certificates"
+	volumeMountNameCABundleDebianFamily        = "debian-family-cabundle"
+	volumeMountPathCABundleDebianFamily        = "/etc/ssl/certs/ca-certificates.crt"
+	volumeMountNameCABundleFedoraRHEL6         = "fedora-rhel6-cabundle"
+	volumeMountPathCABundleFedoraRHEL6         = "/etc/pki/tls/certs/ca-bundle.crt"
+	volumeMountNameCABundleOpensuse            = "opensuse-cabundle"
+	volumeMountPathCABundleOpensuse            = "/etc/ssl/ca-bundle.pem"
+	volumeMountNameCABundleOpenelec            = "openelec-cabundle"
+	volumeMountPathCABundleOpenelec            = "/etc/pki/tls/cacert.pem"
+	volumeMountPathCABundleCentOSRHEL7File     = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
+	volumeMountNameCABundleAlpine              = "alpine-linux-cabundle"
+	volumeMountPathCABundleAlpine              = "/etc/ssl/cert.pem"
+)
+
+// KubeAPIServer contains functions for a kube-apiserver deployer.
+type KubeAPIServer interface {
+	component.DeployWaiter
+	component.MonitoringComponent
+	// SetSecrets sets the secrets.
+	SetSecrets(Secrets)
+	// SetHealthCheckToken sets the health check token for the kube-apiserver
+	SetHealthCheckToken(string)
+	// SetShootAPIServerClusterIP sets the Cluster IP of the service of the shoot apiserver
+	SetShootAPIServerClusterIP(string)
+	// SetShootOutOfClusterAPIServerAddress sets the internal API server domain of the Shoot API Server
+	// used by Gardener system components
+	SetShootOutOfClusterAPIServerAddress(string)
+}
+
+// New creates a new instance of DeployWaiter for the kube-apiserver.
+func New(
+	config *gardencorev1beta1.KubeAPIServerConfig,
+	managedSeedAPIServer *gardencorev1beta1helper.ShootedSeedAPIServer,
+	seedClient kubernetes.Interface,
+	gardenClient client.Client,
```

Comment:
Instead of passing the garden client here, can you rather only provide the audit policy data into this function and read it in the botanist?